tla-import/src/main/scala/at/forsyte/apalache/io/json/TlaToJson.scala

```diff
@@ -2,138 +2,154 @@ package at.forsyte.apalache.io.json
 
 import at.forsyte.apalache.tla.lir.values.{TlaBool, TlaDecimal, TlaInt, TlaStr}
 import at.forsyte.apalache.tla.lir._
+import at.forsyte.apalache.tla.lir.io.TypeTagPrinter
 
 /**
  * A semi-abstraction of a json encoder.
  * It is independent of the concrete JsonRepresentation, resp. JsonFactory implementation.
  *
- * Every internal representation object, Class( arg1 = v1, ... ,argN = vN, otherArgs = args) gets encoded as json in the following way:
+ * Every internal representation object, Class( arg1 = v1, ... ,argN = vN, variadicArgs : T* = args) gets encoded as json in the following way:
```

Comment:
I don't think that renaming `otherArgs` to `variadicArgs` helps a lot. Does it really matter at the json level that some args are variadic?

Reply:
Yes, named args get named fields directly, variadic args get a list.

Reply:
why? I think it is a feature of `TlaBuilder`, not the feature of the IR

Reply:
I mean, take `OperEx`, concretely. It always has an `oper` field, so the json also has an `oper` field. On the other hand, it takes 0 or more actual arguments (an `args: TlaEx*` field), depending on the operator, so the only way you can consistently JSON-serialize it is if you turn `args` into a single JSON field, that holds a list (elements of which are recursively serialized TlaEx into JSON).
What's the alternative?

Reply:
Why cannot we write both `oper` and `args` in a single array that is called `args`?

Reply:
Well, it's not so much that we can't, but I don't think we should. For several reasons, first, it diverges from the internal representation, second, it makes deserialization (which we aren't doing now, but probably want to in the future) harder and third, this is conceptually type-inconsistent (JSON isn't really typed, but our intuition is), as this array would mix objects and strings, which I think is bad form.
I'm going to merge the PR as it is, since it's been sitting for a while (and approved), if you feel strongly about this please open an issue.